libraries/AP_Arming/AP_Arming.h

```diff
@@ -224,6 +228,10 @@ class AP_Arming {
     char* aux_auth_fail_msg;    // buffer for holding failure messages
     bool aux_auth_error;        // true if too many auxiliary authorisers
     HAL_Semaphore aux_auth_sem; // semaphore for accessing the aux_auth_state and aux_auth_fail_msg
+
+    // method that was last used for disarm; invalid unless the
+    // vehicle has been disarmed at least once.
+    Method _last_disarm_method = (Method)-1;
```

Comment:
Let's not do this. If you want an init state, then add something to the enum at the end for "Init" or "Unknown". Blindly casting -1 into the enum is just badly behaved, and doesn't help anyone who wants to do a switch on all the value later.

Reply:
Good call.

Reply:
Fixed.